src/pan3d/widgets/color_by.py

```diff
@@ -0,0 +1,346 @@
+                        flat=True,
+                        variant="outlined",
+                        classes="mx-2",
+                        click=reset_color_range,
+                    )
+            # v3.VDivider()
+            with html.Div(classes="mx-2"):
+                html.Img(
+                    src=(self.__preset_image, None),
+                    style="height: 0.75rem; width: 100%;",
+                    classes="rounded-lg border-thin",
+                )
+            v3.VSelect(
+                placeholder="Color Preset",
+                prepend_inner_icon="mdi-palette",
+                v_model=(self.__color_preset, "Fast"),
+                items=(self.__color_presets, list(PRESETS.keys())),
+                hide_details=True,
+                density="compact",
+                flat=True,
+                variant="solo",
+            )
+
+            with v3.VTooltip(
+                text=("`NaN Color (${nan_colors[nan_color]})`",),
+            ):
+                with html.Template(v_slot_activator="{ props }"):
+                    with v3.VItemGroup(
+                        v_model=self.__nan_color,
+                        v_bind="props",
+                        classes="d-inline-flex ga-1 pa-2",
+                        mandatory="force",
+                    ):
+                        v3.VIcon(
+                            "mdi-eyedropper-variant",
+                            classes="my-auto mx-1 text-medium-emphasis",
+                        )
+                        with v3.VItem(
+                            v_for="(color, i) in nan_colors", key="i", value=("i",)
+                        ):
+                            with v3.Template(
+                                raw_attrs=['#default="{ isSelected, toggle }"']
+                            ):
+                                with v3.VAvatar(
+                                    density="compact",
+                                    color=("isSelected ? 'primary': 'transparent'",),
+                                ):
+                                    v3.VBtn(
+                                        "{{ color[3] < 0.1 ? 't' : '' }}",
+                                        density="compact",
+                                        border="md surface opacity-100",
+                                        color=(
+                                            "color[3] ? `rgb(${color[0] * 255}, ${color[1] * 255}, ${color[2] * 255})` : undefined",
+                                        ),
+                                        flat=True,
+                                        icon=True,
+                                        ripple=False,
+                                        size="small",
+                                        click="toggle",
+                                    )
+
+    @property
+    def data_arrays(self):
+        """Get the data arrays available for coloring."""
+        return self.state[self.__data_arrays]
```

Comment:
we still don't know the shape of those arrays. `[{ name, range, location: point|cell|field}]`?

Reply:
Since you don't have the helper taking a vtkDataSet, we can't even guess easily using the source code of that class.